bittensor/_neuron/text/core_server/run.py

```diff
@@ -208,16 +202,28 @@ def synapse_check(synapse, hotkey):
         """
```

Comment:
why is the synapse check not part of the blacklist check? what is the difference between these two?

Reply:
So the blacklist function operates in the interceptor before the message reaches the axon where the synapse checks are meant to protect individual synapses. For instance, a server may want to protect its seq2seq synapse with strong protections compared to its casuallm, since it requires a lot more compute.